pandas/core/indexes/multi.py

```diff
@@ -3377,14 +3405,11 @@ def isin(self, values, level=None):
             return algos.isin(self.values, values)
         else:
             num = self._get_level_number(level)
-            levs = self.levels[num]
-            level_codes = self.codes[num]
+            levs = self.get_level_values(num)
 
-            sought_labels = levs.isin(values).nonzero()[0]
             if levs.size == 0:
-                return np.zeros(len(level_codes), dtype=np.bool_)
-            else:
-                return np.lib.arraysetops.in1d(level_codes, sought_labels)
+                return np.zeros(len(levs), dtype=np.bool_)
+            return levs.isin(values)
```

Comment:
the edits from L3408 down to here look like they are just nice cleanups independent of the rest of this PR.  is that accurate?

Reply:
@jbrockmendel 
For NA values, someone fixes #30677, then more accurate. "Index.isin" has a bug nonetheless in terms of checking NA value can be possible, This is more accurate